### ISSUE TITLE
Permission denied on SSL certificat

### ISSUE THREAD

**Issue by thromera** [Painted-Fox/docker-postgresql#30]
Fresh install, `docker pull paintedfox/postgresql` and 

```
 docker run -d --name="postgresql" \
             -p 127.0.0.1:5432:5432 \
             -v /tmp/postgresql:/data \
             -e USER="super" \
             -e DB="database_name" \
             -e PASS="$(pwgen -s -1 16)" \
             paintedfox/postgresql
```

Here is the log : 

POSTGRES_USER=super
POSTGRES_PASS=xxx
POSTGRES_DATA_DIR=/data
POSTGRES_DB=database_name
Starting PostgreSQL...
Couldn't initialize inotify.  Are you running Linux 2.6.13 or later, and was the
CONFIG_INOTIFY option enabled when your kernel was compiled?  If so, 
something mysterious has gone wrong.  Please e-mail radu.voicilas@gmail.com
 and mention that you saw this message.
2014-09-09 13:07:23 UTC FATAL:  could not access private key file "/etc/ssl/private/ssl-cert-snakeoil.key": Permission denied

**Comment by theshapguy** [comment]
+1

**Comment by alejo90** [comment]
It has something to do with phusion/baseimage. I reverted it to phusion/baseimage:0.9.10 and it worked.

**Comment by vshulyak** [comment]
Ok, found a solution. Build a new image with --no-cache option from Dockerfile. And probably the image needs to be rebuilt? @Painted-Fox ?

Seems like it's a bug with AuFS mentioned by jpetazzo, not sure if it's relevant though.

**Comment by thromera** [comment]
@alejo90 yes, it's the phusion/baseimage. I moved to `phusion/baseimage:0.9.13` and everything works fine. 
@vshulyak I don't know if @Painted-Fox has something to do with his DockerFile, or if it 's a bug caused by phusion/baseimage...

**Comment by mikeywaites** [comment]
Im still seeing this issue using the current build from the docker registry.

```
==> postgres: 2014-09-30 03:32:36 UTC FATAL:  could not access private key file "/etc/ssl/private/ssl-cert-snakeoil.key": Permission denied
==> postgres: POSTGRES_DATA_DIR=/data
==> postgres: POSTGRES_DB=test
==> postgres: Starting PostgreSQL...
==> postgres: Couldn't initialize inotify.  Are you running Linux 2.6.13 or later, and was the
==> postgres: CONFIG_INOTIFY option enabled when your kernel was compiled?  If so,
==> postgres: something mysterious has gone wrong.  Please e-mail radu.voicilas@gmail.com
==> postgres:  and mention that you saw this message.
```

**Comment by ravster** [comment]
+1.  I'm seeing this error too.  

Postgres is starting up when the image is built with the `--no-cache` option, though.

**Comment by balboah** [comment]
+1

**Comment by bananos** [comment]
+1 

```
2014-10-21 20:08:35 UTC FATAL:  could not access private key file "/etc/ssl/private/ssl-cert-snakeoil.key": Permission denied
POSTGRES_USER=super
POSTGRES_PASS=12345
POSTGRES_DATA_DIR=/data
POSTGRES_DB=cldb
Starting PostgreSQL...
2014-10-21 20:08:36 UTC FATAL:  could not access private key file "/etc/ssl/private/ssl-cert-snakeoil.key": Permission denied
```

Here's the original Docker/AuFS bug https://github.com/docker/docker/issues/783

Any workaround?

**Comment by mlvn23** [comment]
I also encountered this issue, and the work-around is to use devicemapper instead of AUFS.

Add this line in /etc/default/docker:

DOCKER_OPTS="--storage-driver=devicemapper"

and restart the docker service.

**Comment by miloskroulik** [comment]
Well, there seems to be problem with devicemapper too, as mentioned in https://github.com/docker/docker/issues/4860. So, it seems, that it would be better to rebuild the image, as mentioned by @Erowlin

**Comment by verdverm** [comment]
I tried using devicemapper as a backend and started running into this issue:
https://github.com/docker/docker/issues/4036

Still have the original AUFS problem w/o the dm backend

**Comment by miloskroulik** [comment]
I've created pull request based on @Erowlin's comment at https://github.com/Painted-Fox/docker-postgresql/pull/38 It works fine.

**Comment by soupdiver** [comment]
it would be nice to have an updated version pushed to the registry

**Comment by oyvindsk** [comment]
+1

**Comment by zachlatta** [comment]
Running into this issue as well.
